app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/actions/AuthenticatedActionWithRetrievedData.scala

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.actions
+
+import java.time.LocalDate
+
+import com.google.inject.{Inject, Singleton}
+import play.api.Configuration
+import play.api.mvc._
+import uk.gov.hmrc.auth.core.retrieve.{~, Name => RetrievalName, _}
+import uk.gov.hmrc.auth.core.{ConfidenceLevel, _}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.config.ErrorHandler
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.{DateOfBirth, Email, NINO, Name}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.repos.SessionStore
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.util.toFuture
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.HeaderCarrierConverter
+
+import scala.concurrent.{ExecutionContext, Future}
+
+final case class AuthenticatedRequestWithRetrievedData[A](nino: NINO,
+                                                          name: Name,
+                                                          dateOfBirth: DateOfBirth,
+                                                          email: Option[Email],
+                                                          request: MessagesRequest[A])
+  extends WrappedRequest[A](request)
+
+@Singleton
+class AuthenticatedActionWithRetrievedData @Inject()(
+  val authConnector: AuthConnector,
+  val config: Configuration,
+  val errorHandler: ErrorHandler,
+  val sessionStore: SessionStore
+)(implicit val executionContext: ExecutionContext)
+    extends AuthenticatedActionBase[AuthenticatedRequestWithRetrievedData] {
+
+  private def itmpNameToName(itmpName: ItmpName): Option[Name] =
+    itmpName match {
+      case ItmpName(Some(givenName), _, Some(familyName)) => Some(Name(givenName, familyName))
+      case _                                              => None
+    }
+
+  private def nonItmpNameToName(retrievalName: RetrievalName): Option[Name] =
+    retrievalName match {
+      case RetrievalName(Some(name), _) =>
+        name.split(' ').toList.filter(_.nonEmpty) match {
```

Comment:
the only new thing in this file is `.filter(_.nonEmpty)` - I added this in since `"".split(' ') == Array("")`